core/testcontainers/compose/compose.py

```diff
@@ -0,0 +1,419 @@
+import subprocess
+from dataclasses import dataclass, field, fields
+from functools import cached_property
+from json import loads
+from os import PathLike
+from re import split
+from typing import Callable, Literal, Optional, TypeVar, Union
+from urllib.error import HTTPError, URLError
+from urllib.request import urlopen
+
+from testcontainers.core.exceptions import ContainerIsNotRunning, NoSuchPortExposed
+from testcontainers.core.waiting_utils import wait_container_is_ready
+
+_IPT = TypeVar("_IPT")
+
+
+def _ignore_properties(cls: type[_IPT], dict_: any) -> _IPT:
+    """omits extra fields like @JsonIgnoreProperties(ignoreUnknown = true)
+
+    https://gist.github.com/alexanderankin/2a4549ac03554a31bef6eaaf2eaf7fd5"""
+    if isinstance(dict_, cls):
+        return dict_
+    class_fields = {f.name for f in fields(cls)}
+    filtered = {k: v for k, v in dict_.items() if k in class_fields}
+    return cls(**filtered)
+
+
+@dataclass
+class PublishedPort:
+    """
+    Class that represents the response we get from compose when inquiring status
+    via `DockerCompose.get_running_containers()`.
+    """
+
+    URL: Optional[str] = None
+    TargetPort: Optional[str] = None
+    PublishedPort: Optional[str] = None
+    Protocol: Optional[str] = None
+
+
+OT = TypeVar("OT")
+
+
+def one(array: list[OT], exception: Callable[[], Exception]) -> OT:
```

Comment:
I found this functions not so easy to understand: the naming does not imply what it does (e.g. it misses a verb), one has to read at the usage place the second argument/ the exception message to get a feeling what's happening in it. On the other hand, it doesn't really replace much duplicated code: It replaces two three-line `if`-clauses with two one-line `one(...)` calls but you need to convert the straight exception ´raise` into a `lambda`.  But it also has 6 lines of definition.

```
if len(array) != 1: 
    raise Exception(....)
return array[0]
```

vs 

```
return one(array, lambda: Exception(...))
```

Personally, I feel that this is less readable and the longer `if` version is clearer and (with two places) even overall shorter.